# Hypothesis Testing

In [23]:
import pandas as pd
from scipy.stats import ttest_ind

In [25]:
df = pd.read_excel("FEV-data-Excel.xlsx")
df.head()

Car full name  Make                        Model  \
0            Audi e-tron 55 quattro  Audi            e-tron 55 quattro   
1            Audi e-tron 50 quattro  Audi            e-tron 50 quattro   
2             Audi e-tron S quattro  Audi             e-tron S quattro   
3  Audi e-tron Sportback 50 quattro  Audi  e-tron Sportback 50 quattro   
4  Audi e-tron Sportback 55 quattro  Audi  e-tron Sportback 55 quattro   

   Minimal price (gross) [PLN]  Engine power [KM]  Maximum torque [Nm]  \
0                       345700                360                  664   
1                       308400                313                  540   
2                       414900                503                  973   
3                       319700                313                  540   
4                       357000                360                  664   

        Type of brakes Drive type  Battery capacity [kWh]  Range (WLTP) [km]  \
0  disc (front + rear)        4WD                    95.0                438   
1  disc (front + rear)        4WD                    71.0                340   
2  disc (front + rear)        4WD                    95.0                364   
3  disc (front + rear)        4WD                    71.0                346   
4  disc (front + rear)        4WD                    95.0                447   

   ...  Permissable gross weight [kg]  Maximum load capacity [kg]  \
0  ...                         3130.0                       640.0   
1  ...                         3040.0                       670.0   
2  ...                         3130.0                       565.0   
3  ...                         3040.0                       640.0   
4  ...                         3130.0                       670.0   

   Number of seats  Number of doors  Tire size [in]  Maximum speed [kph]  \
0                5                5              19                  200   
1                5                5              19                  190   
2                5                5              20                  210   
3                5                5              19                  190   
4                5                5              19                  200   

   Boot capacity (VDA) [l]  Acceleration 0-100 kph [s]  \
0                    660.0                         5.7   
1                    660.0                         6.8   
2                    660.0                         4.5   
3                    615.0                         6.8   
4                    615.0                         5.7   

   Maximum DC charging power [kW]  mean - Energy consumption [kWh/100 km]  
0                             150                                   24.45  
1                             150                                   23.80  
2                             150                                   27.55  
3                             150                                   23.30  
4                             150                                   23.85  

[5 rows x 25 columns]

In [27]:
df['Engine power [KM]'].isnull().sum()

0

In [38]:
engine_power = df['Engine power [KM]']
engine_power.describe()

count     53.000000
mean     269.773585
std      181.298589
min       82.000000
25%      136.000000
50%      204.000000
75%      372.000000
max      772.000000
Name: Engine power [KM], dtype: float64

#### Test whether there is a significant difference in the average Engine power [KM] of vehicles manufactured by two leading manufacturers i.e. Tesla and Audi. 

In [40]:
# Filter data for Tesla and Audi
tesla_df = df[df['Make'] == 'Tesla']
audi_df = df[df['Make'] == 'Audi']

# Extract Engine power [KM] for both manufacturers
tesla_power = tesla_df['Engine power [KM]']
audi_power = audi_df['Engine power [KM]']

In [44]:
t_stat, p_value = ttest_ind(tesla_power, audi_power)

print(f"T-statistic: {t_stat:.3f}")
print(f"P-value: {p_value:.3f}")

T-statistic: 1.702
P-value: 0.117


In [46]:
# Interpretation
alpha = 0.05  # Significance level
if p_value < alpha:
    print("Reject the null hypothesis: There is a significant difference in average engine power.")
else:
    print("Fail to reject the null hypothesis: No significant difference in average engine power.")

Fail to reject the null hypothesis: No significant difference in average engine power.


#### Since the p-value (0.117) is greater than the significance level (α=0.05), we fail to reject the null hypothesis . 
#### This means there is no statistically significant difference in the average engine power of Tesla and Audi vehicles.
### Insights :
#### Tesla and Audi have comparable average engine power.
#### The slight difference in means is not statistically significant.
### Recommendations :
#### Customers should evaluate other factors beyond engine power.
#### Manufacturers should innovate in areas like range, efficiency, and features to stand out.